Enhance password-only handling in Telnet service

s-b-repo opened the pull request:
Key ImprovementsAutomatic detection of password-only mode: After Telnet negotiation, checks banners for "password:" (or variants) without "login:" or "username:". Skips sending username if password-only detected — directly sends the password. More robust prompt detection: Added "pin:" and better draining of initial banners. Fallback safe: If server re-prompts for login after bad password, it correctly skips the pair. Works with empty login: Still supports manual -l "" for compatibility.

s-b-repo committed:
Enhance password-only handling in Telnet service

Key ImprovementsAutomatic detection of password-only mode: After Telnet negotiation, checks banners for "password:" (or variants) without "login:" or "username:".
Skips sending username if password-only detected — directly sends the password.
More robust prompt detection: Added "pin:" and better draining of initial banners.
Fallback safe: If server re-prompts for login after bad password, it correctly skips the pair.
Works with empty login: Still supports manual -l "" for compatibility.

s-b-repo committed:
Enhance Telnet module with failure detection and prompts

Password-only mode: Automatically detected and skips sending username.
Extensive failure patterns: All provided indicators merged into is_failure() for reliable invalid credential detection.
Multilingual & variant prompts: Covers "password:", "passwd:", "пароль:", "contraseña:", etc.
Robust banner draining in service_telnet to ensure clean state before cracking.
Cleaner send logic and consistent handling of re-prompts.
Enhanced usage message explaining new features.

s-b-repo commented:
Password-only mode: Automatically detected and skips sending username.
Extensive failure patterns: All provided indicators merged into is_failure() for reliable invalid credential detection.
Multilingual & variant prompts: Covers "password:", "passwd:", "пароль:", "contraseña:", etc.
Robust banner draining in service_telnet to ensure clean state before cracking.
Cleaner send logic and consistent handling of re-prompts.
Enhanced usage message explaining new features.

vanhauser-thc commented on hydra-telnet.c:
```diff
@@ -5,111 +5,210 @@ extern char *HYDRA_EXIT;
+        strstr(buf, "pwd:") != NULL || strstr(buf, "pin:") != NULL ||
+        strstr(buf, "пароль:") != NULL || strstr(buf, "contraseña:") != NULL ||
+        strstr(buf, "enter password") != NULL || strstr(buf, "password for") != NULL) {
+      password_prompt_seen = 1;
```
please re-add `strstr(buf, "ennwort") != NULL` and also "asscode"

s-b-repo replied:
sure

s-b-repo replied:
just at work

s-b-repo replied:
is this fine

s-b-repo committed:
Update hydra-telnet.c

s-b-repo commented:
Changes made:

    1.Re-added hydra_strcasestr(buf, "ennwort") (German "Kennwort") in all password prompt checks
    2.Re-added hydra_strcasestr(buf, "asscode") in all password prompt checks
    3.Restored original login→password flow from upstream that was missing (wait for password prompt after sending username)
    4.Fixed strlen(buffer) + 1 in hydra_send calls to match original
    5.Added !is_failure(buf) check before success detection to prevent false positives
    6.Kept all improvements (failure detection, password-only mode, multilingual support)

vanhauser-thc commented:
looks better, thanks.
